### PR TITLE
(feat) add login feature

### PR DESCRIPTION
**Pull request by Ductuan92**
add login feature.

### CODE REVIEW

File: .phpunit.result.cache [hunk 1]
```diff
@@ -1 +1 @@
...
...1,"Tests\\src\\Validation\\UserLoginValidateTest::testExceptionEmptyUsernamePassword with data set \"happy-case-1\"":0.03,"Tests\\src\\Validation\\UserLoginValidateTest::testValidateUserName":0,"Tests\\src\\Validation\\UserLoginValidateTest::testValidateUserName with data set \"happy-case-1\"":0.001,"Tests\\src\\Validation\\UserLoginValidateTest::testExceptionValidateUserName with data set \"happy-case-1\"":0.001,"Tests\\service\\LoginServiceTest::testLogin":0.018,"Tests\\Validation\\UserRequestValidationTest::testValidateEmptyUserNamePassword with data set \"happy-case-1\"":0.002,"Tests\\model\\UserTest::testGetId with data set \"happy-case-1\"":0.001,"Tests\\model\\UserTest::testGetUserName with data set \"happy-case-1\"":0.001,"Tests\\model\\UserTest::testGetPassword":0.002,"Tests\\model\\UserTest::testGetPassword with data set \"happy-case-1\"":0.001,"Tests\\model\\UserTest::testEmail":0.002,"Tests\\model\\UserTest::testEmail with data set \"happy-case-1\"":0.001,"Tests\\Validation\\UserLoginVerifyTest::testVerifyPassword with data set \"happy-case-1\"":0.074,"Tests\\Validation\\UserRequestValidationTest::testValidateEmptyUserNamePassword with data set \"happy-case-2\"":0.001}}
```
Comment (bangdinhnfq) [inline .phpunit.result.cache]: Please add this file to gitignore

File: composer.json [hunk 2]
```diff
@@ -11,7 +11,8 @@
   },
   "require": {
     "vlucas/phpdotenv": "^5.4",
-    "ext-pdo": "*"
+    "ext-pdo": "*",
+    "composer/xdebug-handler": "^3.0"
```
Comment (bangdinhnfq) [inline composer.json]: Should put this package to require-dev

File: src/App/Route.php [hunk 3]
```diff
@@ -88,6 +89,8 @@ public function getUri(): string
      */
     public function getControllerClassName(): string
     {
+        if(empty($this->controllerClassName))
```
Comment (bangdinhnfq) [inline src/App/Route.php]: Please follow code standard https://www.php-fig.org/psr/psr-2/#51-if-elseif-else

File: src/App/Route.php [hunk 4]
```diff
@@ -96,6 +99,8 @@ public function getControllerClassName(): string
      */
     public function getActionName(): string
     {
+        if(empty($this->actionName))
```
Comment (bangdinhnfq) [inline src/App/Route.php]: Please follow code standard https://www.php-fig.org/psr/psr-2/#51-if-elseif-else

File: src/Controller/LoginController.php [hunk 5]
```diff
@@ -51,20 +48,26 @@ public function index(): void
...
-        $userTransfer = $userRequest->getUser();
-        $_SESSION['userID'] = $userTransfer->getId();
+        //validate request
+        $message = $this->userRequestValidation->validateEmptyUserNamePassword($this->userLoginRequest);
+        if ($message != [])
+            View::render('login', $message);
 
-        View::render('index');
+        $user = $this->loginService->Login($this->userLoginRequest);
+        if ($user)
```
Comment (bangdinhnfq) [inline src/Controller/LoginController.php]: How about the case user is empty?

File: src/Validation/UserLoginVerify.php [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+{
+
+    /**
+     * @param $dataPassword
+     * @param $passwordRequest
+     * @return array|string[]
+     */
+    public function verifyPassword($dataPassword, $passwordRequest): bool
+    {
+        //$passwordHash = password_hash($passwordRequest, PASSWORD_DEFAULT);
```
Comment (bangdinhnfq) [inline src/Validation/UserLoginVerify.php]: Please remove unused code

File: src/Validation/UserLoginVerify.php [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+        if(password_verify($passwordRequest,$dataPassword))
+            return true;
+        else
```
Comment (bangdinhnfq) [inline src/Validation/UserLoginVerify.php]: https://www.php-fig.org/psr/psr-2/#51-if-elseif-else

File: src/Validation/UserRequestValidation.php [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+class UserRequestValidation
+{
+    /**
+     * @param $userRequest
+     * @return array|string[]
+     */
+    public function validateEmptyUserNamePassword($userRequest): array
+    {
+        if(empty($userRequest->getUserName()) || empty($userRequest->getPassword()))
```
Comment (bangdinhnfq) [inline src/Validation/UserRequestValidation.php]: https://www.php-fig.org/psr/psr-2/#51-if-elseif-else

**Review by bangdinhnfq — CHANGES_REQUESTED** [review]

**Comment by Ductuan92** [comment]
code already merge to master

**Pull request closed without merging**